### ISSUE TITLE
SSLSessionLength Auto-Reconnect between RNFR and RNTO = 503 Bad sequence of commands

### ISSUE THREAD

**Issue by smboyer** [robinrodricks/FluentFTP#1294]
**FTP Server OS:** Unix
**FTP Server Type:** Serv-U FTP Server v15.3.0
**Client Computer OS:** Windows
**FluentFTP Version:** 47.0.0
**Framework:** .NET 6

Guess I have all the luck.  SSLSessionLength is using default setting. The auto-reconnect is happening between the RNFR and RNTO commands resulting in: 503 Bad sequence of commands.

**Logs :**

```
[INF] >         DownloadFile("c:\temp\ftp-test\storage\ANewFile.txt.download", "/Test/ANewFile.txt", Overwrite, Retry, Delete, Throw)
[INF] >         OpenRead("/Test/ANewFile.txt", Binary, 0, 0)
[INF] >         GetFileSize("/Test/ANewFile.txt")
[INF] Command:  SIZE /Test/ANewFile.txt
[INF] Response: 213 1596204 [62ms]
[INF] >         OpenDataStream("RETR /Test/ANewFile.txt", 0)
[INF] >         OpenPassiveDataStream(AutoPassive, "RETR /Test/ANewFile.txt", 0)
[INF] Command:  EPSV
[INF] Response: 229 Entering Extended Passive Mode (|||5005|) [141ms]
[INF] Connecting to IP #1= ***:5005
[INF] Command:  RETR /Test/ANewFile.txt
[INF] Response: 150 Opening BINARY mode data connection for ANewFile.txt (1596204 Bytes). [131ms]
[WRN] SSL Buffering disabled because of .NET 5.0 and later
[INF] FTPS authentication successful, lib = .NET SslStream, cipher suite = Tls12 (Aes256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, 44550, 256) [127ms]
[INF] Response: 226 Transfer complete. 1,596,204 bytes transferred. 1,749.49 KB/sec. [1.182s]
[INF] >         GetChecksum("/Test/ANewFile.txt")
[INF] Command:  XMD5 /Test/ANewFile.txt
[INF] Response: 250 6FEC8E7BEDB1090FE31D7901B5761608 [67ms]
[INF] File Verification: PASS
[INF] >         MoveFile("/Test/ANewFile.txt", "/Test/Archive/ANewFile.txt", Overwrite)
[INF] >         FileExists("/Test/ANewFile.txt")
[INF] Command:  SIZE /Test/ANewFile.txt
[INF] Response: 213 1596204 [64ms]
[INF] >         FileExists("/Test/Archive/ANewFile.txt")
[INF] Command:  SIZE /Test/Archive/ANewFile.txt
[INF] Response: 550 /Test/Archive/ANewFile.txt: No such file. [67ms]
[INF] >         Rename("/Test/ANewFile.txt", "/Test/Archive/ANewFile.txt")
[INF] Command:  RNFR /Test/ANewFile.txt
[INF] Response: 350 File or directory exists, ready for destination name. [62ms]
[WRN] Reconnect needed due to max SslSessionLength reached on control connection (SslSessionLength: 751)
[INF] Command stashed: RNTO /Test/Archive/ANewFile.txt
[INF] >         Connect(True)
[WRN] Reconnect (Count: 2)
[INF] Connecting to IP #1= ***:21
[INF] Response: 220 Serv-U FTP Server v15.3.0 ready... [201ms]
[INF] Status:   Detected FTP server: ServU
[INF] Command:  AUTH TLS
[INF] Response: 234 AUTH command OK. Initializing SSL connection. [51ms]
[WRN] SSL Buffering disabled because of .NET 5.0 and later
[INF] FTPS authentication successful, lib = .NET SslStream, cipher suite = Tls12 (Aes256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, 44550, 256) [125ms]
[INF] Command:  USER ***
[INF] Response: 331 User name okay, need password. [144ms]
[INF] Command:  PASS ***
[INF] Response: 230 User logged in, proceed. [77ms]
[INF] Command:  PBSZ 0
[INF] Response: 200 PBSZ command OK. Protection buffer size set to 0. [105ms]
[INF] Command:  PROT P
[INF] Response: 200 PROT command OK. Using private data connection. [90ms]
[INF] Text encoding: System.Text.UTF8Encoding+UTF8EncodingSealed
[INF] Command:  OPTS UTF8 ON
[INF] Response: 200 OPTS UTF8 is set to ON. [61ms]
[INF] Command:  SYST
[INF] Response: 215 UNIX Type: L8 [56ms]
[INF] Command:  TYPE I
[INF] Response: 200 Type set to I. [59ms]
[INF] >         SetWorkingDirectory("/")
[INF] Command:  CWD /
[INF] Response: 250 Directory changed to / [59ms]
[INF] Command:  PWD
[INF] Response: 257 "/" is current directory. [61ms]
[INF] 
[INF] Executing stashed command
[INF] 
[INF] Command:  RNTO /Test/Archive/ANewFile.txt
[INF] Response: 503 Bad sequence of commands. [73ms]
[INF] >         Dispose()
[INF] Command:  QUIT
[INF] Response: 221 Goodbye, closing session. [63ms]

```

**Comment by FanDjango** [comment]
There is a list of commands that initiate a sequence that should not be interrupted by an automatic reconnect. I'll add RNFR.

**Comment by FanDjango** [comment]
This should be in the next Nuget then or pickup the current master.

As a workaround right now, you could issue a bogus `PASV` command manually yourself in your code prior to the RNTO, do the RNTO and then a bogus `RETR` command with an invalid filename.

**Comment by FanDjango** [comment]
@smboyer Released as 47.1.0 - should not restart at that exact moment any more.

**Comment by smboyer** [comment]
I can confirm this resolved the issue.  SslSessionLength increased from 751 to 752 before the reconnection attempt.

Log:
```

[INF] Command:  RNFR /Test/ANewFile.txt
[INF] Response: 350 File or directory exists, ready for destination name. [60ms]
[INF] Command:  RNTO /Test/Archive/ANewFile.txt
[INF] Response: 250 RNTO command successful. [93ms]
[INF] >         DownloadFile("c:\temp\ftp-test\storage\AnotherFile.txt.download", "/Test/AnotherFile.txt", Overwrite, Retry, Delete, Throw)
[INF] >         OpenRead("/Test/AnotherFile.txt", Binary, 0, 0)
[INF] >         GetFileSize("/Test/AnotherFile.txt")
[WRN] Reconnect needed due to max SslSessionLength reached on control connection (SslSessionLength: 752)

```
